app/models/task.py

```diff
@@ -1,5 +1,32 @@
 from app import db
+from app.models import goal
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, default= None) #something about has to be null
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable = True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+
+        task_info= {
+            "id": self.task_id,
+            "title" : self.title,
+            "description" : self.description,
+            "is_complete" : self.completed_at is not None # might break code
+        }
+        if self.goal_id:
+            task_info["goal_id"] = self.goal_id
+
+        return task_info
+
+    @classmethod
+    def from_dict(cls, task_data):
+        return cls(
+            title = task_data["title"],
+            description = task_data["description"],
+            completed_at = task_data.get("completed_at", None)
```

Comment:
```suggestion
            completed_at = task_data.get("completed_at")
```
Great usage of the `.get()` method! You can omit the `None` parameter because the default behavior of `.get()` is to return `None` if the key was not found.